graph/layout/eades_example_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package layout
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"gonum.org/v1/gonum/graph/simple"
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/vg"
+)
+
+func ExampleEadesR2() {
+
+	g := makeCompleteGraph(6)
+
+	// Graph layout.
+	// Explained in "A heuristic for graph drawing",
+	// Congressus numerantium 42:149-160.
+	// Experiment with these values for more/less nodes.
+	eades := EadesR2{Repulsion: 1, Rate: 0.05, Updates: 30, Theta: 0.2}
+
+	// Contains graph, layout and updater function.
+	optimizer := NewOptimizerR2(g, eades.Update)
+
+	// Reposition nodes until eades.Updates == 0
+	// by calling layout updater in Update method.
```

Comment:
This is not quite true. `Update` returns whether there is more work to do; both `Updates > 0` and nodes have moved and are finitely located.